run-tests.pl

```diff
@@ -525,9 +525,40 @@ sub _run_test
    $t->start;
    $f_start->done;
 
+   my ( $success, $reason ) = _run_test0( $t, $test, \@req_futures );
+
+   Future->needs_all( map {
+      if( is_Fixture( $_ ) and $_->teardown ) {
+         $_->teardown->( $_ );
+      }
+      else {
+         ();
+      }
+   } @requires )->get;
+
+   if( $success ) {
+      $proven{$_} = PROVEN for @{ $test->proves // [] };
+      $t->pass;
+   }
+   else {
+      $t->fail( $reason );
+   }
+}
+
+# Helper for _run_test. Waits for the req_futures to become ready, then runs
+# the test itself and waits for the test to complete.
+#
+# returns a pair [$res, $reason] where $res is one of:
```

Comment:
Best to document that with `(parens)` so the reader doesn't think it returns a two-element arrayref.

Reply:
done in b8a39cb